Validate and refactor SP-Responsive-UI.js

populationtwo opened the pull request:
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| New sample? | no |
| Related issues? | SP-Responsive-UI.js |
#### What's in this Pull Request?

Hi,

This PR is mainly code improvements. It's a copy of another PR I made on PnP-Tools https://github.com/OfficeDev/PnP-Tools/pull/79

msftclas commented:
Hi __@populationtwo__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, MSBOT;

VesaJuvonen commented:
Hi Edwin,
thanks for the submission. Could you re-target this to Dev, so that we can get it processed. We only do master merge on monthly basis under the PnP core component repository. Thx.

populationtwo commented:
@VesaJuvonen I have updated the PR, along with this [PR](https://github.com/OfficeDev/PnP-Tools/pull/79)

Thanks

msftclas commented:
Hi __@populationtwo__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, MSBOT;

PaoloPia commented on Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js:
```diff
@@ -1,117 +1,121 @@
+
+                // embedding and loading of all necessary CSS files and JS libraries
+                var currentScriptUrl = $('#PnPResponsiveUI').attr('src');
+                if (currentScriptUrl != "undefined") {
```
Hi Edwin,
There is still one issue in this line. Could you please fix it before we can evaluate and eventually merge this PR?
Thanks.

populationtwo replied:
Hi @PaoloPia 

I'm sorry, what's the issue with that line? It was `undefined` but then as suggested on this PR, I changed it to `"undefined"`.

Thanks

VesaJuvonen replied:
Hi Edwin, 
could you please come back on this or we'll need to close the PR. Thx.

PaoloPia commented:
Hi Edwin,
We're evaluating your PR, as you saw in my previous comment. However, we would like to understand better the overall goal. In the PR description, you refer to a bug that you fixed. Can you be more specific about what you fixed and where?
Thanks.

VesaJuvonen commented:
Hi Edwin,
just to follow up on this. We did check the implementation and it looks pretty nice, but we could not really see what is that exactly fixing or what was the actual change from behavior perspective. Could you provide more details on what does this fixes compared to previous version other than changes in the implementation style in CSS / JS files. thx for the additional details advance.

populationtwo commented:
@PaoloPia My mistake. There was no bug, it was mainly code improvements. Fixed some issues I found on JSHint and refactored sideNavToggle.

Since this is an exact duplicate of my PR [here](https://github.com/OfficeDev/PnP-Tools/pull/79), should I copy the PR description from that page? I'm not sure if I should put the same description on both PR.

@VesaJuvonen 
There is no change from behaviors perspective. I basically run the SP-Responsive-UI.js through JShint to detect errors and potential problems. Let me know if you'd like to see a more detailed description from what I have on [here](https://github.com/OfficeDev/PnP-Tools/pull/79).

Thanks!

PaoloPia commented:
Hi Edwin,
We really appreciate your contribution and your efforts. I manually merged some of your proposed changes in order to make proper testing. You can see the PR here (https://github.com/SharePoint/PnP-Sites-Core/pull/857). Thus, I will close this one, but you will be credited in the upcoming monthly community call. Thanks.
